src/common/syncjournaldb.cpp

```diff
@@ -1990,24 +1990,45 @@ void SyncJournalDb::markVirtualFileForDownloadRecursively(const QByteArray &path
     query.exec();
 }
 
-PinState SyncJournalDb::pinStateForPath(const QByteArray &path)
+PinState SyncJournalDb::rawPinStateForPath(const QByteArray &path)
 {
     QMutexLocker lock(&_mutex);
     if (!checkConnect())
-        return PinState::Unspecified;
+        return PinState::Inherited;
```

Comment:
I'm wondering if we should not reply with an error there.

Reply:
I wondered the same thing. From reading other functions in SyncJournalDb it seemed that we assume checkConnect() errors during sync are unlikely enough to not be handled consistently.

It'd be easy to error if pinStateForPath() fails to access the db. I've added that.